src/System.IO.Ports/src/System/IO/Ports/SerialStream.Unix.cs

```diff
@@ -595,18 +595,29 @@ private void EnsureIOLoopRunning()
             }
         }
 
-        private void FinishPendingIORequests(Func<Exception> createException = null)
+        private void FinishPendingIORequests()
         {
-            createException = createException ?? InternalResources.FileNotOpenException;
+            while (_readQueue.TryDequeue(out SerialStreamIORequest r))
+            {
+                r.Complete(InternalResources.FileNotOpenException());
+            }
 
+            while (_writeQueue.TryDequeue(out SerialStreamIORequest r))
+            {
+                r.Complete(InternalResources.FileNotOpenException());
+            }
+        }
+
+        private void FinishPendingIORequestsWithIOException(Interop.ErrorInfo error)
+        {
             while (_readQueue.TryDequeue(out SerialStreamIORequest r))
             {
-                r.Complete(createException());
+                r.Complete(Interop.GetIOException(error));
             }
 
             while (_writeQueue.TryDequeue(out SerialStreamIORequest r))
             {
-                r.Complete(createException());
+                r.Complete(Interop.GetIOException(error));
             }
         }
```

Comment:
Rather than duplicating this method, how about?
```C#
private void FinishPendingIORequestsWithIOException(Interop.ErrorInfo? error = null)
{
    for (int i = 0; i < 2; i++)
    {
        ConcurrentQueue<SerialStreamIORequest> q = i == 0 ? _readQueue : _writeQueue;
        while (q.TryDequeue(out SerialStreamIORequest r))
        {
            r.Complete(error.HasValue ? Interop.GetIOException(error.Value) : InternalResources.FileNotOpenException());
        }
    }
}
```
?

Reply:
The abstraction will be slightly weird since `InternalResources.FileNotOpenException()` is an ObjectDisposedException but I guess there will be a trade-off somewhere in here anyway